bindings/java/src/main/com/apple/foundationdb/FDBTransaction.java

```diff
@@ -428,9 +428,27 @@ protected FutureResults getRange_internal(
 					" -- range get: (%s, %s) limit: %d, bytes: %d, mode: %d, iteration: %d, snap: %s, reverse %s",
 				begin.toString(), end.toString(), rowLimit, targetBytes, streamingMode,
 				iteration, Boolean.toString(isSnapshot), Boolean.toString(reverse)));*/
-			return new FutureResults(
-				Transaction_getRange(getPtr(), begin.getKey(), begin.orEqual(), begin.getOffset(),
-									 end.getKey(), end.orEqual(), end.getOffset(), rowLimit, targetBytes,
+
+			/*
+			 * correctness note: KeySelector.exact() isn't a viable start key, because it
```

Comment:
Could we define `KeySelector.exact()` to return a key selector that is equivalent to `firstGreaterOrEqual`, except with the extra `isExact` flag set? If you did that, then I think `getRange` doesn't have to change anything and `getKey` doesn't get any more difficult.

Reply:
One way or the other, you run into a problem with either `getRange` or `getKey`. If you define `exact` as `firstGreaterOrEqual`, then `getKey(exact)` will return incorrectly (as your note above mentions), but if you define it as I do here, then `getKey` works but `getRange` requires additional work. 

I think it's more-or-less arbitrary which way we go with it (and I'd be perfectly fine swapping), but we either need special behavior in `getKey()` or in `getRange()`

Reply:
I think `getKey` returns the wrong result regardless as currently implemented, but whether you use `lastLessOrEqual` or `firstGreaterOrEqual`, the fixed implementation would be the same.